File: manual/code/test_suites/cpp/FunctionBuildTest/tests/test_matplotlib_scatter.cpp
#include <gtest/gtest.h>
#include "../src/function_matplotlib_scatter.cpp"

class TestScatterPlotGenerator : public ::testing::Test {
};

TEST_F(TestScatterPlotGenerator, TestValidInput) {
    std::string result = ScatterPlotGenerator::create_scatter_plot("1,2,3", "4,5,6");
    EXPECT_EQ(result, "Plot saved as 'scatter_plot.png'.");
}

TEST_F(TestScatterPlotGenerator, TestEmptyInput) {
    std::string result = ScatterPlotGenerator::create_scatter_plot("", "");
    EXPECT_EQ(result, "Error: x_values and y_values cannot be empty.");
}

TEST_F(TestScatterPlotGenerator, TestMismatchedLengths) {
    std::string result = ScatterPlotGenerator::create_scatter_plot("1,2", "3");
    EXPECT_EQ(result, "Error: x_values and y_values must have the same length.");
}

TEST_F(TestScatterPlotGenerator, TestNonNumericValues) {
    std::string result = ScatterPlotGenerator::create_scatter_plot("1,'2',3", "4,5,6");
    EXPECT_EQ(result, "Error: All x and y values must be numbers.");
}

TEST_F(TestScatterPlotGenerator, TestLargeInput) {
    // 生成1000个值
    std::ostringstream x_stream, y_stream;
    for (int i = 0; i < 1000; ++i) {
        if (i > 0) {
            x_stream << ",";
            y_stream << ",";
        }
        x_stream << i;
        y_stream << (i * 2);
    }
    
    std::string result = ScatterPlotGenerator::create_scatter_plot(x_stream.str(), y_stream.str());
    EXPECT_EQ(result, "Plot saved as 'scatter_plot.png'.");
}

int main(int argc, char **argv) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: manual/data/python/tests/test_matplotlib_scatter.py
import unittest
from function_matplotlib_scatter import ScatterPlotGenerator

class TestScatterPlotGenerator(unittest.TestCase):
    def test_valid_input(self):
        result = ScatterPlotGenerator.create_scatter_plot("1,2,3", "4,5,6")
        self.assertEqual(result, "Plot saved as 'scatter_plot.png'.")

    def test_empty_input(self):
        result = ScatterPlotGenerator.create_scatter_plot("", "")
        self.assertEqual(result, "Error: x_values and y_values cannot be empty.")

    def test_mismatched_lengths(self):
        result = ScatterPlotGenerator.create_scatter_plot("1,2", "3")
        self.assertEqual(result, "Error: x_values and y_values must have the same length.")

    def test_non_numeric_values(self):
        result = ScatterPlotGenerator.create_scatter_plot("1,'2',3", "4,5,6")
        self.assertEqual(result, "Error: All x and y values must be numbers.")

    def test_large_input(self):
        x_values = ",".join(map(str, range(1000)))
        y_values = ",".join(map(str, [val * 2 for val in range(1000)]))
        result = ScatterPlotGenerator.create_scatter_plot(x_values, y_values)
        self.assertEqual(result, "Plot saved as 'scatter_plot.png'.")

if __name__ == "__main__":
    unittest.main()
